pkg/lib/authn/authenticator/password/provider.go

```diff
@@ -99,6 +100,7 @@ func (p *Provider) WithPassword(a *authenticator.Password, password string) (*au
 	}
 
 	newAuthen := p.populatePasswordHash(a, password)
+	newAuthen.ExpireAfter = spec.ExpireAfter
```

Comment:
Do not always write this field. If `setPasswordExpired=true` in the Admin API, then we should set it to now. Otherwise, leave the original value unchanged. That is, if the original value is a date that is in the past, it is kept unchanged.

You may encounter a problem that you cannot distinguish between "I want to keep the expire_at unchanged" and "I want to set expire_at to NULL". Think about it.

Reply:
Or should I add an `ResetExpireAfter` flag for existing flows to explicitly reset the expireAFter, then Set will just store latest auth info as is?